src/Dapper.AOT.Analyzers/CodeAnalysis/DapperInterceptorGenerator.cs

```diff
@@ -1167,24 +1169,58 @@ private static void WriteCommandFactory(string baseFactory, CodeWriter sb, IType
         sb.Outdent().NewLine().NewLine();
     }
 
-    private static void WriteRowFactory(CodeWriter sb, ITypeSymbol type, int index)
+    private static void WriteRowFactory(SourceProductionContext context, CodeWriter sb, ITypeSymbol type, int index)
     {
-        var members = Inspection.GetMembers(type).ToImmutableArray();
-        var memberCount = 0;
-        foreach (var member in members)
+        var members = Inspection.GetMembers(type)
+            .Where(member => CodeWriter.IsSettableInstanceMember(member.Member, out _))
+            .ToImmutableArray();
+        var membersCount = members.Length;
+        if (membersCount == 0)
         {
-            if (CodeWriter.IsSettableInstanceMember(member.Member, out _))
-            {
-                memberCount++;
-            }
+            context.ReportDiagnostic(Diagnostic.Create(Diagnostics.UserTypeNoSettableMembersFound, type.Locations.First()));
```

Comment:
should this logic only happen after resolving a constructor? if there's a constructor, the members needn't be settable...

Reply:
Yep, did not think about this scenario. Reporting diagnostic, if members **and** constructor is not found.
`membersCount == 0 && !hasDapperAotCompatibleConstructor`